src/App.tsx

```diff
@@ -1,61 +1,159 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(searchParameter: number | string): User | null {
+  const foundUserById
+  = usersFromServer.find(user => user.id === searchParameter);
+
+  const foundUserByName
+  = usersFromServer.find(user => user.name === searchParameter);
+
+  switch (typeof searchParameter) {
+    case 'string':
+      return foundUserByName || null;
+    default:
+      return foundUserById || null;
+  }
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [isClicked, setIsClicked] = useState(false);
+  const [titleName, setTitleName] = useState('');
+  const [userName, setUserName] = useState('');
+  const [titleBoolean, setTitleBoolean] = useState(false);
+  const [userBoolean, setUserBoolean] = useState(false);
```

Comment:
```suggestion
  const [isTitle, setIsTitle] = useState(false);
  const [isUser, setIsUser] = useState(false);
```

Reply:
BTW do you really need these 2 additional states? You already have userName and titleName. Just check if they exist.